### ISSUE TITLE
`no-matching-overload` false positive for GeoPandas if pandas-stubs is installed

### ISSUE THREAD

**Issue by p-vdp** [astral-sh/ty#3181]
### Summary

I'm getting `no-matching-overload` errors when creating GeoDataFrames in environments that also have pandas-stubs installed. (Though pandas-stubs seems to fix some issues as discussed here: https://github.com/astral-sh/ty/issues/2730)

For example, with this as test.py:
```python
# https://geopandas.org/en/stable/gallery/create_geopandas_from_pandas.html
import geopandas as gpd
import pandas as pd


df = pd.DataFrame(
    {
        "City": ["Buenos Aires", "Brasilia", "Santiago", "Bogota", "Caracas"],
        "Country": ["Argentina", "Brazil", "Chile", "Colombia", "Venezuela"],
        "Latitude": [-34.58, -15.78, -33.45, 4.60, 10.48],
        "Longitude": [-58.66, -47.91, -70.66, -74.08, -66.86],
    }
)

gdf = gpd.GeoDataFrame(
    df, geometry=gpd.points_from_xy(df.Longitude, df.Latitude), crs="EPSG:4326"
)
```

Passes with just GeoPandas:
```shell
uvx --with geopandas ty check test.py
All checks passed!
```

Errors with the addition of pandas-stubs:

```shell
uvx --with geopandas --with pandas-stubs ty check test.py
error[no-matching-overload]: No overload of function `__new__` matches arguments
  --> test.py:15:7
[more diagnostics etc.]
```

### Version

ty 0.0.26 (940305127 2026-03-26)

**Comment by MeGaGiGaGon** [comment]
This is happening because you also need [`types-geopandas`](https://pypi.org/project/types-geopandas/) to make it work:
```
PS ~>uvx --with pandas --with geopandas --with pandas-stubs --with types-geopandas ty check test.py
All checks passed!
```

The reason `types-geopandas` is needed is because both plain `pandas` and `geopandas` use a `__init__` for their `DataFrames`, while `pandas-stubs` uses a `__new__` for it's `DataFrame`. Because of that, since `geopandas` only overrides `__init__` and not `__new__` you get the overload error. `types-geopandas` uses `__new__` like `pandas-stubs`, and so it all works again.

**Comment by p-vdp** [comment]
That seems to fix it!
